Fix compiler warning

MarekKnapek opened the pull request.

FlorianReimold reviewed (changes requested):
Your code may crash (-> undefined behavior) on systems where a char is signed by default, as you must not pass anything to toupper that is not representable by an unsigned char. Thus the old code was correct. If it gave you a warning, you can change the PR and cast to an unsigned char _before_ calling toupper.
https://en.cppreference.com/w/cpp/string/byte/toupper

MarekKnapek committed:
Fix compiler warning about truncation.

Warning C4244: '=': conversion from 'int' to 'char', possible loss of data.

Change lambda parameter type and add explicit cast to avoid possibly confusing implicit cast.

MarekKnapek commented:
Hi @FlorianReimold, I fixed the issue with char / unsigned char you mentioned.
